lib/spec/openapi/index.js

```diff
@@ -26,18 +26,20 @@ module.exports = function (opts, cache, routes, Ref, done) {
     }, ref)
 
     for (const route of routes) {
-      const schema = defOpts.transform
-        ? defOpts.transform(route.schema)
-        : route.schema
+      const transformResult = defOpts.transform
+        ? defOpts.transform(route.schema, route.url)
+        : {}
 
+      const schema = transformResult.schema ?? route.schema
       const shouldRouteHideOpts = {
         hiddenTag: defOpts.hiddenTag,
         hideUntagged: defOpts.hideUntagged
       }
 
       if (shouldRouteHide(schema, shouldRouteHideOpts)) continue
 
-      const url = normalizeUrl(route.url, defOpts.servers, defOpts.stripBasePath)
+      let url = transformResult.url ?? route.url
```

Comment:
Nullish operator add on `node@14` and we support `node@10` currently.
```suggestion
      let url = transformResult.url || route.url
```